src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -79,7 +83,15 @@ public synchronized String getCurrentCommandString() {
             return getCurrentCommand().toString();
     }
 
+    private void storeInHistory(DriverCommand command) {
+        commandHistory.add(command);
+    }
+
     public Publisher getChangePublisher() {
         return changePublisher;
     }
+
+    public synchronized List<DriverCommand> getCommandHistory() {
+        return new ArrayList<>(commandHistory);
+    }
```

Comment:
SRP - history is additional resposibility. Extract it to a separate class. Remember about DIP